File: GitToolBox/src/main/java/zielu/gittoolbox/extension/AutoFetchAllowedEP.java
package zielu.gittoolbox.extension;

import com.intellij.openapi.extensions.ExtensionPointName;
import com.intellij.openapi.project.Project;
import com.intellij.util.xmlb.annotations.Attribute;
import org.jetbrains.annotations.NotNull;
import zielu.intellij.extensions.ZAbstractExtensionPointBean;

public class AutoFetchAllowedEP extends ZAbstractExtensionPointBean {
  public static final ExtensionPointName<AutoFetchAllowedEP>
      POINT_NAME = ExtensionPointName.create("zielu.gittoolbox.autoFetchAllowed");

  @Attribute("provider")
  public String provider;

  public AutoFetchAllowed instantiate(@NotNull Project project) {
    return createInstance(provider, project);
  }
}

File: GitToolBox/src/main/kotlin/zielu/gittoolbox/extension/autofetch/AutoFetchAllowedEP.kt
package zielu.gittoolbox.extension.autofetch

import com.intellij.openapi.project.Project
import com.intellij.util.xmlb.annotations.Attribute
import zielu.intellij.extensions.ZAbstractExtensionPointBean

internal class AutoFetchAllowedEP : ZAbstractExtensionPointBean() {
  @Attribute("provider")
  lateinit var provider: String

  fun instantiate(project: Project): AutoFetchAllowed {
    return createInstance(provider, project)
  }
}
